osDiscovery/osDiscovery_test.go

```diff
@@ -187,12 +196,18 @@ func TestGetArchitecture(t *testing.T) {
 	if _, err := GetArchitecture(); err != ErrUnknownArchitecture {
 		t.Errorf("Expected error \"%s\"; got %s", ErrUnknownArchitecture.Error(), err.Error())
 	}
+
+	// 3: new line
+	if out, _ := GetArchitecture(); out != "ARCH" {
+		t.Errorf("Expected \"arch\"; got %s", out)
+	}
 }
 
 func TestGetKernel(t *testing.T) {
 	s := stubUtils.NewStubs(t,
 		&stubUtils.CmdStub{Cmd: "uname", Args: []string{"-r"}, Output: []byte("KERN")},
-		&stubUtils.CmdStub{Cmd: "uname", Args: []string{"-r"}, Err: stubUtils.OhNoErr})
+		&stubUtils.CmdStub{Cmd: "uname", Args: []string{"-r"}, Err: stubUtils.OhNoErr},
+		&stubUtils.CmdStub{Cmd: "uname", Args: []string{"-r"}, Output: []byte("KERN\n")})
```

Comment:
Try to group code together by content. It would make more sense to group together valid responses and then do the error check at the end.